src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -513,6 +514,13 @@ class spell_warl_drain_soul : public SpellScriptLoader
                 });
             }
 
+            bool CheckProc(ProcEventInfo& eventInfo)
+            {
+                // Drain Soul's proc tries to happen each time the warlock lands a killing blow on a unit while channeling.
+                // Make sure that dying unit is afflicted by the caster's Drain Soul debuff in order to avoid a false positive.
+                return eventInfo.GetProcTarget()->GetAuraApplicationOfRankedSpell(SPELL_WARLOCK_DRAIN_SOUL, GetCaster()->GetGUID()) != 0;
```

Comment:
is there a chance for GetCaster() to return null at this point ?

Reply:
I'm not entirely familiar with the scripting system, so I'm can't say it wouldn't with 100% certainty. I definitely got caught up in the idea of a smooth-looking one-liner.

I'll make the following adjustments:

* Switch from `GetCaster()` to `eventProc.GetActor()`.
* Store pointers for `GetProcTarget` and `GetActor` instead of the one-liner.
* Wrap the final `if` in a safety check for the above stored pointers.